### ISSUE TITLE
Light DOM styles per component

### ISSUE THREAD

**Issue by LeaVerou** [shoelace-style/webawesome#301]
Several use cases keep cropping up where components need to style their descendants or other elements in the light DOM.

We should use a static property to specify light DOM styles per component (`lightStyle`? analogously to `shadowStyle`) that are added to its parent shadow root as well as any ancestor ones, all the way to the top. This would only happen once per component type per shadow root/document.

Naively sprouting styles can interfere with DOM diffing (thanks @KonnorRogers for pointing that out), but using `document.adoptedStyleSheets` [works around that](https://codepen.io/leaverou/pen/GgKjbdQ/ab3ec8672be7e16dbb1117c339f163c0). This won't work in SSR, so we'd need to detect this and add a regular `<style>` element in that case.

We should also use this issue to collect a list of the many, MANY use cases that have been cropping up over time. Things like:
- `<wa-page>` should create utility classes such as `mobile-only` etc.
- `<wa-page>` should hide `[data-nav-toggle]` in the desktop view, so that the author doesn't need to do it
- `<wa-page>` should apply 0 spacings to `html, body`
- Several cases where we have to use our own child component instead of allowing arbitrary ones just so we can style it (@claviska must remember the specifics)
- Basically any case where we need to apply styles to slotted elements, and some for `:host`
